src/h5web/dataset-visualizer/VisDisplay.tsx

```diff
@@ -24,11 +24,16 @@ function VisDisplay(props: Props): JSX.Element {
   // https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops
   if (activeVis !== prevVis) {
     setPrevVis(activeVis);
-    setMapperState(
-      activeVis === Vis.Line
-        ? [...range(0, datasetDims.length - 1).fill(0), 'x']
-        : undefined
-    );
+    if (activeVis === Vis.Line) {
+      setMapperState([...range(datasetDims.length - 1).fill(0), 'x']);
+    } else if (
+      activeVis === Vis.Heatmap ||
+      (activeVis === Vis.Matrix && datasetDims.length > 2)
+    ) {
+      setMapperState([...range(datasetDims.length - 2).fill(0), 'y', 'x']);
```

Comment:
For now, the mapping of `Matrix` only displays the datasets of dimensions above 2D as 2D matrices (no display as 1D vector yet).

Reply:
I think this should go in `VIS_DEFS`, perhaps a new method called `getInitialMapperState()` declared for each type of visualization.